### PR TITLE
Repo cleanup: consolidate/rename folders, add/update READMEs

### PR DESCRIPTION
**Pull request by leemthompo**
ℹ️ Don't worry about the files changed number, it's ~90% renaming 😄 

- Updates/adds READMEs to all main sections: 
  - main README
  - example-apps
  - notebooks
- Renames/consolidates example apps
- Copyedits existing instructions

Probably didn't catch everything here, but can circle back for smaller, targeted updates.

### CODE REVIEW

File: example-apps/ElasticGPT_Plugin/README.md
Comment (chriscressman) [inline example-apps/ElasticGPT_Plugin/README.md:1]: We could clarify how our two docs GPT example apps relate. This one is for those who want to use the ChatGPT UI, while the other is for those who want to use the GPT API with their own UI.

**Review by chriscressman — COMMENTED** [review]
I took a look at all the READMEs. Looks like a good round of general cleanup to me :smile: 

May want to rename some of the example app directories to give them more generic names that describe the type of app or use case. The notebook directories are a good example of this.

File: example-apps/ElasticGPT_Plugin/README.md
Comment (leemthompo) [reply example-apps/ElasticGPT_Plugin/README.md:1]: Good point, I can clarify this in the individual readmes the overall `example-apps` readme.

File: example-apps/ElasticGPT_Plugin/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+- The plugin then returns the document body and URL to ChatGPT, which uses this information to craft its response.
+- Deployment of the plugin on Google Cloud Platform (GCP) is demonstrated using Cloud Run.
+- Users can install the plugin in ChatGPT and query Elasticsearch data to enhance ChatGPT's knowledge and functionality.
+
+For full details refer to the original Elastic Blog Post:
+
+- [ChatGPT and Elasticsearch: A plugin to use ChatGPT with your Elastic data](https://www.elastic.co/blog/chatgpt-elasticsearch-plugin-elastic-data)
+
+
+> ℹ️ **Note**:
```
[the remark this answers is not in the document]
Comment (leemthompo) [reply example-apps/ElasticGPT_Plugin/README.md:17]: > move up higher in the hierarchy to cover everything?

🎯

**Review by joemcelroy — APPROVED** [review]
looks great! could you rebase?

File: example-apps/Workplace-Search/README.md [hunk 2]
```diff
@@ -6,11 +6,12 @@ This is a sample app that combines Elasticsearch and OpenAI to create a semantic
 
 ## 1. Download the Project
 
-Download the project from Github and extract the python-flask-example folder.
+Download the project from Github and extract the Workplace-Search folder.
 
 ```bash
 curl https://codeload.github.com/elastic/elasticsearch-labs/tar.gz/main | \
-tar -xz --strip=2 elasticsearch-labs-main/esre-examples/python-flask-example
+tar -xz --strip=2 elasticsearch-labs-main/esre-examples/Workplace-Search
```
Comment (leemthompo) [inline example-apps/Workplace-Search/README.md:13]:
```suggestion
tar -xz --strip=2 elasticsearch-labs-main/example-apps/Workplace-Search
```

File: example-apps/ElasticGPT_Plugin/README.md
Comment (leemthompo) [reply example-apps/ElasticGPT_Plugin/README.md:1]: Added to follow-up PR

File: example-apps/ElasticGPT_Plugin/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (leemthompo) [reply example-apps/ElasticGPT_Plugin/README.md:17]: Added to follow-up PR

**Pull request merged**